core/src/main/scala/ai/lum/odinson/OdinsonIndexWriter.scala

```diff
@@ -81,9 +101,16 @@ class OdinsonIndexWriter(
     addDocuments(block)
   }
 
-  def commit(): Unit = writer.commit()
+  def commit(): Unit = {
+    writer.flush()
+    writer.commit()
+  }
 
   def close(): Unit = {
```

Comment:
@marcovzla - can you review the changes to this method? The Lucene methods don't appear to support overwriting or merging existing files, so the only way I could get it to work was to rewrite the file each time...